pystac_client/item_search.py

```diff
@@ -17,11 +17,11 @@
 from pystac_client.stac_api_io import StacApiIO
 
 if TYPE_CHECKING:
-    from pystac_client import client
+    from pystac_client import client as _client
```

Comment:
Could we just import `Client` directly?

Reply:
I tried that, and I got a type warning at the place it was string quoted. I'll try it again -- it could have been caused by some other issue that we've now resolved.

Reply:
I was able to resolve it now, and it's fixed in a later PR